### ISSUE TITLE
Checkpoint 2 ... ish?

### ISSUE THREAD

**Issue by billyJoePiano** [billyJoePiano/TenaPull#5]
@pawaitemadisoncollege

Most of my work so far has been around the Nessus API, figuring out how it is structured (see the .json files in the 'notes' folder for examples) and then how to de-serialize it for the local DB.  The 'entity' and 'persistence' classes are very much a work in progress.  My goal here was to avoid writing a separate Dao for each POJO class.  A number of the DB tables are also going to be simple "lookup" style tables (an auto_increment ID and a single varchar field, to avoid highly-redundant varchar columns in the main tables).  This is why I created the special interface and Dao for those lookup POJOs.

I glanced ahead at the material coming up, and it seems I may be getting ahead of myself with JSON and some of that other POJO/DAO stuff.  The data structure coming out of Nessus is pretty sophisticated, and it will need to be filtered because not all of that data is actually of interest to us.  Basically, there is no way I'd be able to get full CRUD implemented at this point... and I think that was reflected in my original timeline.  The goal right now is understanding the API structure, and starting to setup some of the framework for processing it.

Also, I'm honestly not even sure that those entity/persistence classes will compile right now.  I'm unable to test that out because I'm still struggling greatly with the IntelliJ IDE configurations, getting it to correctly recognize packages, dependencies, and whatnot.  I will definitely need your help with this again.  I am using IntelliJ on my host OS (as opposed to the Ubuntu VM) and even just trying to do a local build creates all sorts of problems.   I finally got it to recognize the Hibernate annotations after quite a struggle and restarting it several times.  The current problem, which wasn't an issue until after Hibernate was recognized, is that the IDE is not recognizing classes from the main.nessusData.entity package in both the persistence package and even from within other classes also in the entity package (!!!!)    Very frustrating, and makes it impossible to actually test any of this code.  I feel like I am still spending about as much time fighting IntelliJ over these configurations as I am actually writing code, sadly.

In any case, I hope this is sufficient for Checkpoint 2, given the nature of my project.  Please let me know your thoughts.

Thanks,

Bill

**Comment by billyJoePiano** [comment]
Update -- I cloned a fresh copy of the repo and was able to get IntelliJ recognizing the dependencies and packages correctly.  Had to make a few changes, but it appears that the code is at least compiling now.  Haven't had a chance to run any tests on it yet.

**Comment by pawaitemadisoncollege** [comment]
Hi @billyJoePiano 

After we chatted on Monday, I took a look at your Dao - it's very similar to the GenericDao provided in the demo project and is a good start! I think the lookup dao code could be generalized and rolled right to your Dao class as this functionality could be used by any entity. Review the provided GenericDao for inspiration around single and multiple column searches (it uses a map when searching by multiple columns and values).

With regard to the api - this week's readings, watchings and activities should be a big help moving this along. If you complete the exercise using the Nessus api, you'll have 
1. the classes created from JSON response using the RoboPojoGenerator      
2. the dao to call the api endpoint and parse the response into objects
3. unit tests to verify 2 is working properly

Here's what's needed yet for Checkpoint 2:
- [x] ERD - does not have to be final, of course, but a first cut.
- [x] one table created and unit tested for full crud with hibernate (include a screenshot showing tests passing). Even if you won't need full crud functionality in your application, there is no harm in having unit tests for this - it ensures the annotations and hibernate config are set up correctly, particularly as your being working with related tables.
- [x] update time log (last entry 3/1)

Regarding fighting with IntelliJ - it certainly isn't a perfect tool, but I do think sometimes you head down a path that may not follow the recommended technique or process  - not using maven for the build/deploy, and not using the provided GenericDao, for example.  It's certainly ok to explore other options, but, as you've found, this can slow progress!
